### ISSUE TITLE
Illegal division by zero at pgcluu line 2696, <GEN123> line 1.

### ISSUE THREAD

**Issue by lyrixx** [darold/pgcluu#39]
I generated the .csv with:

```bash
perl pgcluu_collectd `pwd`/report --no-sysinfo --disable-sar -U sensiolabs -h MYHOST --verbose -d postgres
```

BTW, `--no-sysinfo` is not enought, I add to add `--disable-sar` to make it work

**Comment by darold** [comment]
Thanks for the report. Please download latest code, last commit 11de2e5 migth solves this issue. Maybe you have killed pgcluu_collectd before the next loop so autodetect interval from the data file is not possible. The patch force an interval of 1 second in that case.

--no-sysinfo is only used to not collect kernel and and other system configuration information. --disable-sar should be used if you do not have sar installed on your system.

Best regards,

**Comment by lyrixx** [comment]
> Maybe you have killed pgcluu_collectd before the next loop so autodetect interval from the data file is not possible. The patch force an interval of 1 second in that case.

It's exactly what I did. ;)

Thanks for your answer about sar. But If I don't want sysinfo, I mean I want only db info. And so sar is not needed, right?

Another thing: The debug is not easy, because "you" don't output psql error.
I just get "FATAL: psql error" ([see](https://github.com/darold/pgcluu/blob/master/pgcluu_collectd#L1005))

But, because of the lack of information, I had to execute:
```
sed -i 's/psql error\.\\n"/psql error.\\n".__LINE__/' pgcluu_collectd
```

and because, I did not see the $?, my sed did not work ;)

I finally dumped `PSQL_PROG` to understand what happend.

**Comment by darold** [comment]
You might want to look at your postgresql log file to see what's happen, but here I think this is a connection issue because it fail at the very first query "select version();"  You might want to check your pg_hba.conf and see what your server log report.

Sar is not mandatory but you have so specify the --disable-sar command line option.

**Comment by darold** [comment]
But i'm agree that here using:

    &dprint("FATAL: psql error. $pg_ver\n");

will help to have the error message :-) I will fix that.

**Comment by lyrixx** [comment]
I don't have log (easilly) of my psql, because I'm using RDS.

Anyway I fixed all my issues.

I will close this thread.

And BTW, with RDS, I miss lot of data :( I will see what I can do ;)

**Comment by darold** [comment]
Fixed in commit 5468852

**Comment by darold** [comment]
I've never tried pgcluu with RDS, let us know if there's any issue.

Regards,
